mvrx/src/main/kotlin/com/airbnb/mvrx/lifecycleAwareLazy.kt

```diff
@@ -16,7 +14,7 @@ private object UninitializedValue
  */
 @RestrictTo(RestrictTo.Scope.LIBRARY)
 @SuppressWarnings("Detekt.ClassNaming")
-class lifecycleAwareLazy<out T>(private val owner: LifecycleOwner, initializer: () -> T) : Lazy<T>, Serializable {
+class lifecycleAwareLazy<out T>(owner: LifecycleOwner, initializer: () -> T) : Lazy<T>, Serializable, DefaultLifecycleObserver {
```

Comment:
@marukami suggestion

    class lifecycleAwareLazy<out F>(
        owner: LifecycleOwner,
        initializer: () -> F
    ) : Lazy<F>, Serializable, DefaultLifecycleObserver {

I don't know what is the _correct_ Kotlin style according Google/JetBrains when there are tons of parameters and interfaces, but this is the way I cheat the linter and keep things readable (well, at least it does not complain 😉)

Reply:
> I don't know what is the correct Kotlin style according Google/JetBrains

There isn't one 😀 I find it a bit hard as the project style XML seems to be at odds with the detetkt rules and AS keeps differing from the projects actual rules. 

I only removed the private scope access. I guess we can update the style to match things while we are here.

Reply:
When I made this PR this was the project style. After #387 is merged I'm going to rebase and format to match the updated styles.